src/resolvers/dns.rs

```diff
@@ -404,11 +434,18 @@ pub struct DnsResolverConfig {
     /// Default: 5
     pub max_dns_concurrency: usize,
 
-    /// How long should we wait before re-querying DNS servers?
+    /// How long should we wait before re-querying DNS servers, if we received
+    /// at least one backend record?
     ///
     /// Default: 60 seconds
     pub query_interval: Duration,
 
+    /// How long should we wait before re-querying DNS servers, if we received
+    /// no backend records from a DNS server?
+    ///
+    /// Default: 10 seconds
+    pub query_retry: Duration,
```

Comment:
Bikeshedding welcome on this name.

Reply:
I think this name overlaps too much with `query_interval` - it's more verbose, but I think `query_retry_if_no_records_found` is *really* explicit that this is for an edge-case condition.